airbyte-webapp/src/pages/connections/ConnectionTransformationPage/DbtCloudTransformationsCard.tsx

```diff
@@ -25,6 +25,50 @@ interface DbtJobListValues {
   jobs: DbtCloudJob[];
 }
 
+const DbtCloudErrorBoundary = class extends React.Component {
+  state = { error: null, displayMessage: null };
+
+  // TODO parse the error to determine if the source was the upstream network call to
+  // the dbt Cloud API. If it is, extract the `user_message` field from dbt's error
+  // response for display to user; if not, provide a more generic error message. If the
+  // error was *definitely* not related to the dbt Cloud API, consider reraising it.
+  static getDerivedStateFromError(error: Error) {
+    // TODO I'm pretty sure I did not correctly mock the exact error response format.
+    // eslint-disable-next-line
+    const displayMessage = (error?.message as any)?.status?.user_message;
+    return { error, displayMessage };
+  }
+
+  componentDidCatch(error: Error) {
+    console.log(error);
+  }
```

Comment:
How about using our `AppMonitoringService` to track when errors catch here? Right now we would not have any idea about how often this is happening, since the error is just handled in the UI. With `trackError` we can log occurrences in datadog. There are examples of how to do this with an error boundary [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/pages/SpeakeasyRedirectPage/SpeakeasyRedirectPage.tsx) and [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-webapp/src/components/common/ApiErrorBoundary/ApiErrorBoundary.tsx#L83)